automotive/odCANDataStructureGenerator/org.opendavinci.candatamodel/src/org/opendavinci/generator/CANDataModelGenerator.xtend

```diff
@@ -50,46 +52,77 @@ class CANDataModelGenerator implements IGenerator {
+        val mapOfDefinedCANMessages = collectDefinedCANMessages(resource.allContents.toIterable.filter(typeof(CANMessage)))
+        
 		// First, extract all CAN signals from .can file.
-		val mapOfDefinedCANSignals = collectDefinedCANSignals(resource.allContents.toIterable.filter(typeof(CANSignal)))
+		//val mapOfDefinedCANSignals = collectDefinedCANSignals(resource.allContents.toIterable.filter(typeof(CANSignal)))
 		
 		// Next, extract all .odvd header names from the 'using' statement.
 		val odvdIncludedFiles = extractOdvdHeaders(resource.allContents.toIterable.filter(typeof(ODVDFile)))
 		
 		// needed for the "super" header and source files
 		var ArrayList<String> includedClasses=new ArrayList<String>
-		for (e : resource.allContents.toIterable.filter(typeof(CANSignalMapping))) {
+		for (e : resource.allContents.toIterable.filter(typeof(CANMessageMapping))) {
 			includedClasses.add(e.mappingName.toString().replaceAll("\\.", "/"))
 		}
 		
 		fsa.generateFile("include/GeneratedHeaders_" + generatedHeadersFile + ".h", generateSuperHeaderFileContent(generatedHeadersFile, includedClasses, odvdIncludedFiles))
 		fsa.generateFile("src/GeneratedHeaders_" + generatedHeadersFile + ".cpp", generateSuperImplementationFileContent(generatedHeadersFile, includedClasses))
 		
-		var ArrayList<CANSignalTesting> tests=new ArrayList<CANSignalTesting>(resource.allContents.toIterable.filter(typeof(CANSignalTesting)).toList);
+		var ArrayList<CANMessageTesting> tests=new ArrayList<CANMessageTesting>(resource.allContents.toIterable.filter(typeof(CANMessageTesting)).toList);
 		
 		// Next, generate the code for the actual mapping.
-		for (e : resource.allContents.toIterable.filter(typeof(CANSignalMapping))) {
-			fsa.generateFile("include/generated/" + e.mappingName.toString().replaceAll("\\.", "/") + ".h", generateHeaderFileContent(generatedHeadersFile, odvdIncludedFiles, e))
+		for (e : resource.allContents.toIterable.filter(typeof(CANMessageMapping))) {
+			fsa.generateFile("include/generated/" + e.mappingName.toString().replaceAll("\\.", "/") + ".h", 
+			    generateHeaderFileContent(generatedHeadersFile, odvdIncludedFiles, e))
 			fsa.generateFile("src/generated/" + e.mappingName.toString().replaceAll("\\.", "/") + ".cpp", 
-				generateImplementationFileContent(e, "generated", mapOfDefinedCANSignals))
-			fsa.generateFile("testsuites/" + e.mappingName.toString().replaceAll("\\.", "_") + "TestSuite.h", generateTestSuiteContent(generatedHeadersFile, odvdIncludedFiles, e, tests, mapOfDefinedCANSignals))
-			fsa.generateFile("uppaal/generated/" + e.mappingName.toString().replaceAll("\\.", "/"), generateUPPAALFileContent(e, mapOfDefinedCANSignals))
+				generateImplementationFileContent(e, "generated", mapOfDefinedCANMessages))
+			fsa.generateFile("testsuites/" + e.mappingName.toString().replaceAll("\\.", "_") + "TestSuite.h", 
+			    generateTestSuiteContent(generatedHeadersFile, odvdIncludedFiles, e, tests, mapOfDefinedCANMessages))
```

Comment:
@fgiaimo could you please check with @HangYin2015 if the UPPAAL part from the generator could be removed entirely to improve readability?